Dockerfile-tests

```diff
@@ -0,0 +1,46 @@
+# We build an image with bee-queue's node_modules and all the source
+# files.  We jump through some Docker FROM hoops to avoid having the
+# bloat of the g++ compiler and other tools in the final image.
+
+FROM node:lts-alpine as beequeue-base
+
+# Info, and add bash, which we need in the final image
+RUN true \
+  && echo -n "node " && node --version \
+  && echo -n "npm " && npm --version \
+  && echo -n "apk " && apk --version \
+  && apk update \
+  && apk add --no-cache bash \
+  && rm -r /usr/share/man \
+  && true
+
+FROM beequeue-base as beequeue-node_modules
+
+# Ensure that tools for npm are available (for node-gyp in particular)
+# to build node_modules
```

Comment:
Are there any dependencies of bee-queue that actually require this?

Reply:
`node-gyp` is the bad boy here.  Has been a PITA for years.  Something in the set of bee-queue (dev?) packages needs it.  We could get more nuanced if necessary, but this is a simple solution right now.

Reply:
Strange - that isn't happening for me. It's happening for you on alpine or just on mac?

Reply:
I was a bit terse yesterday, long day.  It's somewhat complicated.

`nody-gyp` is used by npm when C/C++ libraries need to be compiled for npm modules.  I've seen it used for packages that need to wrangle bits and bytes to construct and/or unpack binary C structs to send over the wire, e.g. Sequelize to talk to SQLservers.  I'm guessing that the `redis` library wants to do the same thing to talk to Redis -- just guessing.

AFAICT, npm has several options about how to proceed in a given `npm install` environment:
- pull in pre-compiled libraries for the platform
- pull in node-gyp tools and compile the C/C++ code 
- fall back to a Javascript implementation
 
Using Alpine brings lots of these issues to the fore.  It can be a bit tedious, but it does help shed light on some subtle background matters.  Using Debian or Ubuntu won't have this friction...

If you look at the Hello World example PR you'll see that I don't pull in the build system, and the example works.  If you remove the `--quiet` option you'll see that there are numerous errors emitted by npm regarding not finding Python and that its "not OK".  Yet, the install succeeds and the example works.

If I add `apk add` Python, the build then emits errors about `make` not being present, yet the example works.  And if I add `make` then it emits errors about not having `g++`, etc.

I'm guessing that in the face of these errors, the build for `redis` falls back to a native JS implementation -- again, just guessing.

For the Hello World example this underlying turmoil is irrelevant, and whatever npm has installed is good enough.

But, for starting on a Docker configuration that wants to be reliable vis-a-vis Bee-Queue testing and performance, I want to use a robust build chain.  So, that's what's in this PR.  In the long run (if there is one) we would certainly want to have the testing vet both versions of the `redis` runtime (as Travis does with various Node versions).  I would also want to build/test with a Debian base image for similar exposure to different underlying tool chains.

Reply:
And, I'm stubborn about continuing to use a frictional build chain if it helps us isolate and correct race conditions that are not manifest in the standard CI chain.

Reply:
>I'm guessing that the redis library wants to do the same thing to talk to Redis -- just guessing.

That was once the case - redis used a library called `hiredis` - but that hasn't been the case for a long time.

I think a general (minor) goal of this project is to not have any C/C++ dependencies, but I don't know that that needs to block this PR.